### PR TITLE
[SDK-CORE] Load w/ metadata

### LINKED ISSUE #1112: [SDK-CORE] Use metadata for framework initialization
## Overview

The `create` method in `Framework.ts` can use the metadata package to retrieve all the necessary contract addresses for initialization. 
We can also consider lazy loading of the `SuperToken` class. 

### Non-Breaking
- [ ] We keep `Famework.create` as returning a promise, but when a resolver address is passed as an option or something other than v1 is passed as the `protocolReleaseVersion` then we use the existing code path, else we use the metadata package.

### Breaking
- ~[ ] `Framework.create` no longer returns a promise and we initialize the contracts with the metadata addresses. If it is in a local environment, the metadata addresses will not exist for this and we will need to add a new function `initialize` which runs the existing framework creation code which loads the resolver.~  
  - **`Framework.create` must return a promise to check that the chainId from the provider is the same as the inputted chainId.**

### PR DESCRIPTION
**Pull request by 0xdavinchee**
* Initialize Framework class w/ metadata for supported networks
* Continue using resolver/superfluidLoader for non-supported networks

### CODE REVIEW

**Review by kasparkallas — COMMENTED** [review]
Changelog?

File: packages/sdk-core/README.md [hunk 1]
```diff
@@ -50,6 +50,13 @@ SDK-Core is in early active development and can have breaking releases without w
 
 # Prerequisites
 
+> NOTE: You need to have graphql installed as a dependency in order to use SDK-Core:
```
Comment (kasparkallas) [inline packages/sdk-core/README.md]: Also `ethers`, I guess...

Comment (0xdavinchee) [reply packages/sdk-core/README.md]: ah true

File: packages/sdk-core/src/constants.ts [hunk 2]
```diff
@@ -32,29 +31,41 @@ const subgraphReleaseTag =
...
+const getAddressesData = (chainId: number): NetworkData => {
     const networkData = metadata.networks.find((x) => x.chainId === chainId);
-    if (!networkData)
-        return {
-            subgraphAPIEndpoint: "",
-            networkName: "",
-            resolverAddress: "",
-            nativeTokenSymbol: "",
-        };
+    if (!networkData) return EMPTY_NETWORK_DATA;
```
Comment (kasparkallas) [inline packages/sdk-core/src/constants.ts]: Why not just return `null` or `undefined`? It's better to be explicit than have empty strings and weird empty objects be passed around...

File: packages/sdk-core/README.md [hunk 3]
```diff
@@ -50,6 +50,13 @@ SDK-Core is in early active development and can have breaking releases without w
 
 # Prerequisites
 
+> NOTE: You need to have graphql and ethers installed as a dependency in order to use SDK-Core:
+
+```bash
+yarn install --D graphql ethers
```
Comment (kasparkallas) [inline packages/sdk-core/README.md]: I don't think `--D` is necessary here...

**Review by kasparkallas — CHANGES_REQUESTED** [review]

File: packages/sdk-core/README.md (hunk 3 above)
Comment (0xdavinchee) [reply packages/sdk-core/README.md]: si

File: packages/sdk-core/src/constants.ts (hunk 2 above)
Comment (0xdavinchee) [reply packages/sdk-core/src/constants.ts]: hmmm touché

**Comment by 0xdavinchee** [comment]
@kasparkallas pls re-review, requested changes have been applied

File: packages/sdk-core/src/constants.ts [hunk 4]
```diff
@@ -32,29 +31,41 @@ const subgraphReleaseTag =
     (globalThis.process && globalThis.process.env.SUBGRAPH_RELEASE_TAG) ||
     DefaultSubgraphReleaseTag.value;
 
-const getResolverData = (chainId: number): IResolverData => {
+export const EMPTY_NETWORK_DATA = {
```
Comment (kasparkallas) [inline packages/sdk-core/src/constants.ts]: Not necessary anymore?

Comment (0xdavinchee) [reply packages/sdk-core/src/constants.ts]: still used in `Framework.ts`, can change the logic in there slightly to disappear this object.

Comment (kasparkallas) [reply packages/sdk-core/src/constants.ts]: Please do :wink:

Comment (0xdavinchee) [reply packages/sdk-core/src/constants.ts]: donezo

**Review by kasparkallas — APPROVED** [review]

**Pull request merged**